File: Array_Carvans.py
# ----------------------------------------------------------
# CodeChef Problem: CARVANS
# Concept: Array traversal and speed comparison
# ----------------------------------------------------------
 
# Input: Number of test cases
t = int(input("Enter number of test cases: "))
 
for _ in range(t):
    # Input: Number of cars
    n = int(input("\nEnter number of cars: "))
 
    # Input: Speeds of each car separated by spaces
    speeds = list(map(int, input("Enter speeds of cars: ").split()))
 
    # If speeds entered are LESS than number of cars → ERROR
    if len(speeds) != n:
        print("⚠️ Please enter exactly", n, "speeds!")
        continue  # skip this test case
 
    count = 1  # First car always maintains speed
 
    for i in range(1, n):
        if speeds[i] > speeds[i - 1]:
            speeds[i] = speeds[i - 1]
        else:
            count += 1
 
    print("✅ Number of cars maintaining speed:", count)
    

File: Array_Carvans.cpp
// Problem 1 (Array)
// Program: CARVANS (CodeChef)

 
#include <iostream>
using namespace std;
 
int main() {
    int t;
    cin >> t; // Number of test cases
 
    while (t--) {
        int n;
        cin >> n; // Number of cars
 
        int speed[n];
        for (int i = 0; i < n; i++) {
            cin >> speed[i];
        }
 
        int count = 1; // The first car is always moving at max allowed speed
        int min_speed = speed[0];
 
        for (int i = 1; i < n; i++) {
            if (speed[i] <= min_speed) {
                count++;
                min_speed = speed[i];
            }
        }
 
        cout << count << endl;
    }
 
    return 0;
}